File: VB/Step 3 - creating view 2 (PersonsViewModel and PersonsView)/ViewModels/ViewModelBase.vb
Imports System.ComponentModel

Namespace DX_MVVM.ViewModels
    Public Class ViewModelBase
        Implements INotifyPropertyChanged

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Protected Overridable Sub OnPropertyChanged(ByVal propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub
    End Class
End Namespace

File: CS/Step 3 - creating view 2 (PersonsViewModel and PersonsView)/ViewModels/ViewModelBase.cs
using System.ComponentModel;

namespace DX_MVVM.ViewModels {
    public class ViewModelBase : INotifyPropertyChanged {
        public event PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged(string propertyName) {
            if(PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}
